rust/threshold-mock/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+use candid::{CandidType, Principal};
+use ic_stable_structures::memory_manager::{MemoryId, MemoryManager, VirtualMemory};
+use ic_stable_structures::{DefaultMemoryImpl, StableBTreeMap};
+use rand::SeedableRng;
+use rand_chacha::ChaCha20Rng;
+use std::cell::RefCell;
+use std::time::Duration;
+
+type Memory = VirtualMemory<DefaultMemoryImpl>;
+
+pub mod ecdsa;
+pub mod schnorr;
+pub mod vetkd;
+
+thread_local! {
+    static RNG: RefCell<Option<ChaCha20Rng>> = const { RefCell::new(None) };
+    static MEMORY_MANAGER: RefCell<MemoryManager<DefaultMemoryImpl>> =
+        RefCell::new(MemoryManager::init(DefaultMemoryImpl::default()));
+    static CALL_COUNTS: RefCell<StableBTreeMap<String, u64, Memory>> = RefCell::new(
+            StableBTreeMap::init(
+                MEMORY_MANAGER.with_borrow(|m| m.get(MemoryId::new(1))),
+            )
+        );
+}
+
+const SEEDING_INTERVAL: Duration = Duration::from_secs(3600);
```

Comment:
what does seeding mean and why is it needed?

Reply:
Seeding the underlying RNG, the code was taken 1:1 from https://internetcomputer.org/docs/current/developer-docs/security/security-best-practices/misc#2-using-raw_rand-as-seed-for-a-psuedo-random-number-generator-prng. I'll improve then naming to make clear the seeding is for the RNG.

Reply:
https://github.com/dfinity/chainkey-testing-canister/pull/1